File: framework/src/main/java/com/framework/utils/UIUtils.java
package com.framework.utils;


import android.content.ClipboardManager;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.ArrayRes;
import android.support.annotation.NonNull;
import android.support.design.widget.Snackbar;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import com.framework.App;


/**
 * by y on 2016/7/26.
 */
public class UIUtils {


    public static Context getContext() {
        return App.getInstance().getApplicationContext();
    }

    public static String getString(int id) {
        return getContext().getString(id);
    }

    public static String[] getStringArray(@ArrayRes int id) {
        return getContext().getResources().getStringArray(id);
    }

    public static View getAdapterView(ViewGroup viewGroup, int id) {
        return LayoutInflater.from(viewGroup.getContext()).inflate(id, viewGroup, false);
    }

    public static void startActivity(@NonNull Class<?> clz) {
        Intent intent = new Intent(getContext(), clz);
        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
        getContext().startActivity(intent);
    }

    public static void startActivity(@NonNull Class<?> clz, Bundle bundle) {
        Intent intent = new Intent(getContext(), clz);
        intent.putExtras(bundle);
        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
        getContext().startActivity(intent);
    }


    public static void toast(Object o) {
        Toast.makeText(getContext(), o.toString(), Toast.LENGTH_SHORT).show();
    }

    public static void snackBar(View view, Object object) {
        Snackbar.make(view, object + "", Snackbar.LENGTH_SHORT).show();
    }

    public static void copy(String content) {
        ClipboardManager cm = (ClipboardManager) getContext().getSystemService(Context.CLIPBOARD_SERVICE);
        cm.setText(content);
    }
}

File: framework/src/main/java/com/framework/utils/UIUtils.kt
package com.framework.utils


import android.content.ClipboardManager
import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.support.annotation.ArrayRes
import android.support.design.widget.Snackbar
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast

import com.framework.App


/**
 * by y on 2016/7/26.
 */
object UIUtils {


    val context: Context
        get() = App.instance.applicationContext

    fun getString(id: Int): String {
        return context.getString(id)
    }

    fun getStringArray(@ArrayRes id: Int): Array<String> {
        return context.resources.getStringArray(id)
    }

    fun getAdapterView(viewGroup: ViewGroup, id: Int): View {
        return LayoutInflater.from(viewGroup.context).inflate(id, viewGroup, false)
    }

    fun startActivity(clz: Class<*>) {
        val intent = Intent(context, clz)
        intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK
        context.startActivity(intent)
    }

    fun startActivity(clz: Class<*>, bundle: Bundle) {
        val intent = Intent(context, clz)
        intent.putExtras(bundle)
        intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK
        context.startActivity(intent)
    }


    fun toast(o: Any) {
        Toast.makeText(context, o.toString(), Toast.LENGTH_SHORT).show()
    }

    fun snackBar(view: View, `object`: Any) {
        Snackbar.make(view, `object`.toString() + "", Snackbar.LENGTH_SHORT).show()
    }

    fun copy(content: String) {
        val cm = context.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager

        cm.text = content
    }
}
